### DISCUSSION TITLE
TypeError: ConfusionMatrix.__new__() missing 1 required positional argument: 'task'

### DISCUSSION THREAD

**Discussion by KaisarJamil** [mrdbourke/pytorch-deep-learning#483] [Q&A]
I tried to plot the confusion matrix in "03_pytorch_computer_vision" session following below:
`from torchmetrics import ConfusionMatrix
from mlxtend.plotting import plot_confusion_matrix

2. Setup confusion instance and compare predictions to targets
confmat = ConfusionMatrix(num_classes=len(class_names))
confmat_tensor = confmat(preds=y_pred_tensor,
                         target=test_data.targets)

3. Plot the confusion matrix
fig, ax = plot_confusion_matrix(
    conf_mat=confmat_tensor.numpy(), # matplotlib likes working with numpy
    class_names=class_names,
    figsize=(10, 7)
)`

**But it occurred an error :** 

`---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
[<ipython-input-68-3194dd00e4e9>](https://localhost:8080/#) in <cell line: 5>()
      3 
      4 # 2. Setup confusion instance and compare predictions to targets
----> 5 confmat = ConfusionMatrix(num_classes=len(class_names))
      6 confmat_tensor = confmat(preds=y_pred_tensor,
      7                          target=test_data.targets)

TypeError: ConfusionMatrix.__new__() missing 1 required positional argument: 'task'`

**Comment by IIsameerII** [comment]
on the second step when you initialize the confusion matrix, you need to specify the task. The correct code would be -

`confmat = ConfusionMatrix(task='multiclass',num_classes=len(class_names))`

Go through the confusion matrix doc, for more info
(https://torchmetrics.readthedocs.io/en/stable/classification/confusion_matrix.html)

**Reply by TabotCharlesBessong** [reply]
Thank you, very much for the fix
